frontend/src/Components/Form/DeviceInputConnector.js

```diff
@@ -2,13 +2,13 @@ import PropTypes from 'prop-types';
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
 import { createSelector } from 'reselect';
-import { fetchOptions, clearOptions } from 'Store/Actions/providerOptionActions';
+import { fetchOptions, clearOptions, defaultState } from 'Store/Actions/providerOptionActions';
 import DeviceInput from './DeviceInput';
 
 function createMapStateToProps() {
   return createSelector(
     (state, { value }) => value,
-    (state) => state.providerOptions,
+    (state) => state.providerOptions.getDevices || defaultState,
```

Comment:
Not really following how `getDevices` is used here. I'd also expect it to be `devices` as a property instead of `getDevices` which looks like a function.

Reply:
Currently we only have one dynamic field in indexers. But I considered the possibility of multiple. Which would interfere with eachother since the options would be stored in providerOptions state directly.
by specifying a subsection we can keep them separate.

For devices I can specify the subsection in the UI code. But for newznab the action 'getNewznabCategories' is specified by the field definition.
So I didn't want to have `devices` as subsection and `getNewznabCategories`. 
But I can strip the `get` from the action when generating the section name.